Checkbox Component

dancingshell opened the pull request:
### Summary
Checkbox (new component) 1.0.0
---
* straightforward checkbox component 
~~* use prop `styleVariants={['infoGrid']}` to use with info-grid~~

ben-burlingham commented on checkbox/src/Checkbox.jsx:
```diff
@@ -0,0 +1,63 @@
+    const { styleVariants, stylesheets } = this.props;
+    const styles = composeStyles(baseStyles, [...getStylesheetsFromVariants(styleVariants).concat(stylesheets)]);
+    return (
+      <input
```
I'd recommend using a `<div>` here rather than a native form element, because native form elements are styled differently depending on the browser.  Those styles can't be overwritten at all.

If we are to have a true Procore checkbox, we'll need to style everything about it - borders, check style, etc.

ben-burlingham commented on checkbox/sandbox/sandbox.jsx:
```diff
@@ -0,0 +1,13 @@
+render(
+  <Checkbox
+    checked
+    name='sandboxCheckbox'
```
Is there a need for a `name` prop?

chrisarevir replied:
Form elements require a name prop for submission

dancingshell replied:
Also for the onChange callback passes back `{value, name}`. If you have multiple checkboxes in your form this makes it easy to keep track of which one is changing

ben-burlingham replied:
Makes sense.  We use a `scope` variable in other particles, perhaps we should change them all to `name` for consistency.

ben-burlingham replied:
@chrisarevir yes, but that's only in a case of a `<form>` element with a child `type="submit"` button...an uncommon pattern in client side SPAs.  Also if we're making a custom checkbox so we can style it, rather than be forced to use the ugly operating system level ones, the `submit` check becomes irrelevant.

mcclayton commented on checkbox/src/Checkbox.jsx:
```diff
@@ -0,0 +1,63 @@
+  };
+  constructor(props) {
+    super(props);
+    this.state = { isChecked: this.props.checked };
```
Similar to the Text Input Component, I don't think this should be stateful.

brock-noah commented on checkbox/src/css-overrides/info-grid-checkbox-override.scss:
```diff
@@ -0,0 +1,3 @@
+.container {
+  margin: 0;
```
I'm confused as why this override is necessary. The CSS defined on this component's base styles doesn't specify margin.

dancingshell replied:
this gets rid of the default margin around input checkbox what came from the browser. I'll admit I only tested that in chrome

dancingshell replied:
actually in the new way I'm implementing this, it is not necessary, removing!

dancingshell commented:
* prop `name` -> `scope`
* removed statefulness
* added analytics 
* updated component to not use input styling, remove need for inline variant
![screen shot 2017-10-12 at 1 16 56 pm](https://user-images.githubusercontent.com/6993945/31517100-b664429c-af4f-11e7-86da-68aeb4b262e1.png)

ben-burlingham commented:
Approving to keep them moving, more bugs will be exposed during use and we can iterate on the theory and the new bugs.  Let's get this party started.
